src/integration-test/java/com/mongodb/jdbc/integration/MongoIntegrationTest.java

```diff
@@ -0,0 +1,134 @@
+package com.mongodb.jdbc.integration;
+
+import com.mongodb.jdbc.MongoConnection;
+import com.mongodb.jdbc.MongoDriver;
+import java.io.File;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import org.junit.jupiter.api.Test;
+
+public abstract class MongoIntegrationTest {
+
+    /**
+     * Creates a new connection.
+     *
+     * @param extraProps Extra properties on top of the default ones that the class implementating
+     *     it is using.
+     * @return The connection.
+     * @throws SQLException If the connection can not be created.
+     */
+    public abstract Connection getBasicConnection(Properties extraProps) throws SQLException;
```

Comment:
Why not just make all of these `Connection`s `MongoConnection`s so that we don't need the down casts in the `cleanup`? It will require casting in the overriden child classes, but it's 2 casts versus 3 :)